python/lsst/drp/tasks/assemble_coadd.py

```diff
@@ -379,7 +379,22 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
         else:
             warpRefList = inputData["inputWarps"]
 
-        inputs = self.prepareInputs(warpRefList)
+        # Although psfMatchedWarps are specifically required by only by a
+        # specific set of a subclass, and since runQuantum is not overridden,
+        # the selection and filtering must happen here on a conditional basis.
+        # Otherwise, the elements in the various lists will not line up.
```

Comment:
Might be worth extending this comment to make it clear that this does indeed suggest that  `AssembleCoadd` and `CompareWarpAssembleCoadd` need to be refactored; you could reference DM-38630 as a blocker, if not necessarily a TODO ticket (I'm definitely not suggesting it do that as part of this ticket!).